### ISSUE TITLE
Can't compile

### ISSUE THREAD

**Issue by preovaleo** [gabime/spdlog#98]
Hello,

i just try to compile the example but i obtain an error :
``````````
g++    -c -g -std=c++11 -MMD -MP -MF "build/Debug/GNU-Linux-x86/example.o.d" -o build/Debug/GNU-Linux-x86/example.o example.cpp
In file included from spdlog/details/spdlog_impl.h:30:0,
                 from spdlog/spdlog.h:155,
                 from example.cpp:2:
spdlog/details/registry.h: In instantiation of ‘static spdlog::details::registry_t<Mutex>& spdlog::details::registry_t<Mutex>::instance() [with Mutex = std::mutex]’:
spdlog/details/spdlog_impl.h:37:31:   required from here
spdlog/details/registry.h:161:5: error: conversion from ‘std::nullptr_t’ to non-scalar type ‘std::function<void()>’ requested
     registry_t<Mutex>() = default;
     ^
spdlog/details/registry.h:149:34: note: synthesized method ‘spdlog::details::registry_t<Mutex>::registry_t() [with Mutex = std::mutex]’ first required here 
         static registry_t<Mutex> s_instance;
                                  ^
```````````

Do you have something to help me?

**Comment by gabime** [comment]
which compiler version? os?

**Comment by preovaleo** [comment]
Ubuntu 14.04.2 g++ 4.8.2

**Comment by gabime** [comment]
Try to add  ```-pthread``` flag if this doesn't help then just comment out this line (spdlog/details/registry.h:16)

**Comment by giellamo** [comment]
I've the same problem... the problem is during compilation so no linker error here...

**Comment by gabime** [comment]
Does replacing with ```registry_t<Mutex>(){}``` helps?

**Comment by gabime** [comment]
Fixed in 1dcaa45443015215d329e9837cf6c4473ec9a8ab
